torch/csrc/autograd/profiler.cpp

```diff
@@ -418,9 +485,62 @@ void Event::record(bool record_cuda) {
   cpu_ns_ = getTime();
 }
 
-double Event::cuda_elapsed_us(const Event & e) {
+/* static */ Event Event::fromIValue(const at::IValue& eventIValue) {
+  TORCH_INTERNAL_ASSERT(
+      eventIValue.isList(),
+      "Expected IValue to contain type c10::impl::GenericList");
+  auto ivalues = eventIValue.toList();
+  TORCH_INTERNAL_ASSERT(
+      ivalues.size() >= kEventIValuesSize,
+      "Expected at least ",
+      kEventIValuesSize,
+      " elements to reconstruct Event.");
+
+  Event evt(
+    static_cast<EventKind>(ivalues.get(0).toInt()), // EventKind
+    at::StringView(ivalues.get(1).toStringRef()), // name
+    ivalues.get(2).toInt(), // thread
+    static_cast<at::RecordFunctionHandle>(ivalues.get(3).toDouble()), // handle
+    {}, // TODO: record shapes
```

Comment:
ok, so we won't have shapes for remote events? I guess that's fine for now

Reply:
I'll file issues for all the follow ups to make sure we don't forget.

Reply:
https://github.com/pytorch/pytorch/issues/39969